### PR TITLE
Migrate AWS Lambda guide

### PR DESCRIPTION
**Pull request by jwitz**
This guide is not great... one of the lines used "IIRC" and half of the links in the "Alternatives" section were broken. Worth having @kentdanas review this to see if we should republish at all, and if so what we should do about this after migration.

### CODE REVIEW

**Comment by kwcanuck** [comment]
@jwitz are you merging with the correct branch?

File: guides/aws-lambda.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+---
+title: "Best Practices Calling AWS Lambda from Airflow"
+description: "A few tips, guidelines, and best practices for calling Lambda from Airflow"
+id: aws-lambda
+sidebar_label: AWS Lambda
+---
+
+Follow this guide to learn some best practices for triggering [AWS Lambda](https://goo.gl/zYGM7L) functions directly from Airflow.
```
Comment (kwcanuck) [inline guides/aws-lambda.md:8]:
```suggestion
In this guide you'll learn some best practices for triggering [AWS Lambda](https://goo.gl/zYGM7L) functions directly from Airflow.
```

File: guides/aws-lambda.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+## AWS Lambda basics
+
+AWS Lambda lets you run code without provisioning or managing servers. You can use AWS Lambda to execute code in response to triggers such as a change in data or a particular user action.
+
+It can be directly triggered by a variety of services and can be orchestrated in workflows. Naturally, the latter can effectively be done using Airflow.
+
+In general, Astronomer recommends coupling Airflow with a serverless Lambda framework such as [Serverless](https://serverless.com/), as there is a significant amount of boilerplate code involved in writing, deploying, updating, and maintaining both serverless functions and an API Gateway.
```
Comment (kwcanuck) [inline guides/aws-lambda.md:16]:
```suggestion
Astronomer recommends coupling Airflow with a serverless Lambda framework such as [Serverless](https://serverless.com/), as there is a significant amount of boilerplate code involved in writing, deploying, updating, and maintaining both serverless functions and an API Gateway.
```

File: guides/aws-lambda.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+## Lambda & Airflow
+
+To call an AWS Lambda function in Airflow, you have a few options:
```
Comment (kwcanuck) [inline guides/aws-lambda.md:20]:
```suggestion
To call an AWS Lambda function in Airflow, you have the following options:
```

File: guides/aws-lambda.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+### Invoke a call in Boto3
+
+The simplest way to call an AWS Lambda function in Airflow is to [invoke](https://boto3.readthedocs.io/en/latest/reference/services/lambda.html#Lambda.Client.invoke) it in [Boto3](https://aws.amazon.com/sdk-for-python/) with the [PythonOperator](https://registry.astronomer.io/providers/apache-airflow/modules/pythonoperator).
+
+The [AwsLambdaHook](https://registry.astronomer.io/providers/amazon/modules/awslambdahook) itself uses the [AwsBaseHook](https://registry.astronomer.io/providers/amazon/modules/awsbasehook). This is is a wrapper around the boto3 library, which is the standard way to interact with AWS using Python. After you configure an Airflow connection to AWS, you can use the hook in one of your operators to trigger a particular Lambda function.
```
Comment (kwcanuck) [inline guides/aws-lambda.md:26]:
```suggestion
The [AwsLambdaHook](https://registry.astronomer.io/providers/amazon/modules/awslambdahook) uses the [AwsBaseHook](https://registry.astronomer.io/providers/amazon/modules/awsbasehook). This is is a wrapper around the boto3 library, which is the standard way to interact with AWS using Python. After you configure an Airflow connection to AWS, you can use the hook in one of your operators to trigger a particular Lambda function.
```

File: guides/aws-lambda.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+### Use the SimpleHttpOperator
+
+Another option would be to use the [SimpleHttpOperator](https://registry.astronomer.io/providers/http/modules/simplehttpoperator) in Airflow to hit the Lambda function, much like a REST API.
```
Comment (kwcanuck) [inline guides/aws-lambda.md:30]:
```suggestion
Use the [SimpleHttpOperator](https://registry.astronomer.io/providers/http/modules/simplehttpoperator) in Airflow to run the Lambda function, much like a REST API.
```

Not sure if _run_ is correct, but _hit_ isn't appropriate here.

File: guides/aws-lambda.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+### Trigger AWS Lambda using an AWS API gateway
+
+You can trigger Lambda functions using an [AWS API Gateway](https://docs.aws.amazon.com/apigateway/latest/developerguide/getting-started-with-lambda-integration.html).
+
+You can still use the `SimpleHttpOperator` as a starting point, but you now also have to handle HTTP Authorization.
```
Comment (kwcanuck) [inline guides/aws-lambda.md:36]:
```suggestion
You can still use the `SimpleHttpOperator` as a starting point, but you now also have to handle HTTP authorization.
```
Is it _authorization_, or _authentication_?

**Review by kwcanuck — APPROVED** [review]

**Comment by jwitz** [comment]
Let's hold off on this one until we can at least get an example DAG in here. This isn't up to par with the other guides we're migrating.

**Pull request closed without merging**